app/controllers/providers/proceedings_types_controller.rb

```diff
@@ -33,5 +34,11 @@ def permitted_params
     def proceeding_types
       @proceeding_types ||= ProceedingType.all
     end
+
+    def back_step_path
+      return providers_legal_aid_application_address_selection_path if applicant.address&.lookup_used?
+
+      providers_legal_aid_application_address_path
+    end
```

Comment:
You shouldn't have to create a new method. Just over-ride the `back_step_url` method.
```ruby
def show
  authorize @legal_aid_application
  proceeding_types
end
   .....
def back_step_url
  super unless legal_aid_application.checking_answers?
  return providers_legal_aid_application_address_selection_path if applicant.address&.lookup_used?

  providers_legal_aid_application_address_path
end
```
As the method is already defined as a helper method in `Steppable`, `back_step_path` will be available directly in the views.